plugins/onboarding-backend/src/service/Github.service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from 'axios';
+import { base64YamlToJSON } from './util';
+
+export class GithubService {
+  baseRepo = '';
+  constructor(props: { baseRepo: string }) {
+    this.baseRepo = props.baseRepo;
+  }
+  async getGithubData() {
+    try {
+      const response = await axios(this.baseRepo, {
```

Comment:
Have you thought about using the Octokit library instead of using http requests?